File: PHP/src/aoc_template.php
<?php

namespace sjb\aoc2024;

require 'lib/Util.php';
// require 'lib/Geometry.php';
// require 'lib/Grid.php';

echo "Advent of Code 2024, Day <##>: <##>\n";

$INPUT_DIR = '../Input/';
$INPUT_FILE = 'day<##>_test.txt';
//$INPUT_FILE = 'day<##>_challenge.txt';
$INPUT_INDEX = 0;

$input = read_grouped_input($INPUT_DIR . $INPUT_FILE, $INPUT_INDEX);

$result1 = solvePartOne($input);
echo "Part One: $result1\n";

//$result2 = solvePartTwo($input);
//echo "Part Two: $result2\n";


function solvePartOne(array $input):int {
	return 0;
}

function solvePartTwo(array $input):int  {
	return 0;
}

File: Perl/aoc_template.pl
#!/usr/bin/env perl
use v5.40;
# use feature 'class';
# no warnings qw( experimental::class );

our $directory;
BEGIN { use Cwd; $directory = cwd; }
use lib $directory . '/lib';

use feature 'signatures';
#use Data::Printer;
#use Storable 'dclone';

use AOC::Util;
#use AOC::Geometry;
#use AOC::Grid;

my $INPUT_FILE = 'day<##>_test.txt';
#my $INPUT_FILE = 'day<##>_challenge.txt';
my @input = read_input("../input/$INPUT_FILE");

say "Advent of Code 2024, Day <##>: <##>";

solve_part_one(@input);
#solve_part_two(@input);

exit( 0 );

sub solve_part_one(@input) {

	say "Part One: ";
}

sub solve_part_two(@input) {

	say "Part One: ";
}
